app/models/profiles_search.rb

```diff
@@ -32,6 +32,7 @@ def detailed_search_result
     return Profile.none if @query.values.all? &:blank?
     result = Profile
       .where('city ILIKE :city', city: "%#{@query[:city]}%")
+      .where('country ILIKE :country', country: "%#{[:country]}")
```

Comment:
here is `@query` missing. but then we found out that when you first come to the detailed search page without any search params theer shoul dbe all profiles shown but there are only profiles show that have a country field all other profiles are not shown at all.